### PR TITLE
M2-3051 Duplicated activity miss alert option value

### PR DESCRIPTION
**Pull request by wildkiomi**

### CODE REVIEW

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 1]
```diff
@@ -98,6 +98,34 @@ export const isPerformanceTaskResponseType = (responseType: ItemResponseType) =>
   responseType === ItemResponseType.Flanker ||
   responseType === ItemResponseType.ABTrails;
 
+const getDuplicatedOptionsAndAlertsForSingleAndMultipleSelect = (item?: ItemFormValues) => {
```
Comment (IhorNesterenko99) [inline src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:101]: pls reduce the size of function name

File: src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx [hunk 2]
```diff
@@ -98,6 +98,34 @@ export const isPerformanceTaskResponseType = (responseType: ItemResponseType) =>
...
+  const options = (item?.responseValues as SingleAndMultipleSelectItemResponseValues)?.options?.map(
+    (option: SingleAndMultiSelectOption) => ({
+      ...option,
+      oldId: option.id,
+      id: uuidv4(),
+    }),
+  );
+
+  const alerts = item?.alerts
```
[the remark this answers is not in the document]
Comment (wildkiomi) [reply src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:110]: I need to know an old and new option id for alerts

[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:110]: try something like this:
const getDuplicatedOptionsAndAlerts = (item?: ItemFormValues) => {
  const { responseValues, alerts } = item || {};
  const newIdsObject: Record<string, string> = {};
  const options = (responseValues as SingleAndMultipleSelectItemResponseValues)?.options?.map(
    (option: SingleAndMultiSelectOption) => {
      const id = uuidv4();
      newIdsObject[option.id] = id;

      return { ...option, id };
    },
  );

  const mappedAlerts = alerts
    ? {
        alerts: alerts.map((alert) => ({
          ...alert,
          value: newIdsObject[alert.value ?? ''],
          key: uuidv4(),
        })),
      }
    : {};

  return {
    responseValues: { ...responseValues, options },
    ...mappedAlerts,
  };
};

[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:110]: try this:

`const getDuplicatedOptionsAndAlerts = (item?: ItemFormValues) => {
  const { responseValues, alerts } = item || {};
  const newIdsObject: Record<string, string> = {};
  const options = (responseValues as SingleAndMultipleSelectItemResponseValues)?.options?.map(
    (option: SingleAndMultiSelectOption) => {
      const id = uuidv4();
      newIdsObject[option.id] = id;

      return { ...option, id };
    },
  );

  const mappedAlerts = alerts
    ? {
        alerts: alerts.map((alert) => ({
          ...alert,
          value: newIdsObject[alert.value ?? ''],
          key: uuidv4(),
        })),
      }
    : {};

  return {
    responseValues: { ...responseValues, options },
    ...mappedAlerts,
  };
};`

[the remark this answers is not in the document]
Comment (IhorNesterenko99) [reply src/modules/Builder/pages/BuilderApplet/BuilderApplet.utils.tsx:110]: `const getDuplicatedOptionsAndAlerts = (item?: ItemFormValues) => {
  const { responseValues, alerts } = item || {};
  const newIdsObject: Record<string, string> = {};
  const options = (responseValues as SingleAndMultipleSelectItemResponseValues)?.options?.map(
    (option: SingleAndMultiSelectOption) => {
      const id = uuidv4();
      newIdsObject[option.id] = id;

      return { ...option, id };
    },
  );

  const mappedAlerts = alerts
    ? {
        alerts: alerts.map((alert) => ({
          ...alert,
          value: newIdsObject[alert.value ?? ''],
          key: uuidv4(),
        })),
      }
    : {};

  return {
    responseValues: { ...responseValues, options },
    ...mappedAlerts,
  };
};`

**Review by IhorNesterenko99 — APPROVED** [review]

**Review by kdanilenka32 — APPROVED** [review]

**Review by alex-i-e — APPROVED** [review]

**Pull request merged**